File: src/main/java/org/homework20250105/java/animal/Cat.java
package org.homework20250105.java.animal;

import org.homework20250105.java.animal.base.Animal;
import org.homework20250105.java.animal.base.ManyFeet;

public class Cat extends Animal implements ManyFeet {
    public Cat(String name, Integer age) {
        super(name, age, AnimalEnum.CAT);
    }

    @Override
    public void say() {
        System.out.println("Meow");
    }

    @Override
    public int getFeet() {
        return 4;
    }
}

File: src/main/java/org/homework20250105/kotlin/animal/Cat.kt
package org.homework20250105.kotlin.animal;

import org.homework20250105.kotlin.animal.base.Animal
import org.homework20250105.kotlin.animal.base.ManyFeet

class Cat(name: String, age: Int?): Animal(name, age, AnimalEnum.CAT) , ManyFeet {

    override fun say() {
        println("Meow")
    }

    override fun getFeet(): Int {
        return 4
    }
}